test/integration/narrow/app/fuzzySearchService.test.js

```diff
@@ -116,12 +86,92 @@ describe('testing fuzzySearchService /paymentdata', () => {
       .toThrow('Empty search content')
   })
 
-  test('GET /paymentdata parameter default valus used', async () => {
+  test('GET /paymentdata throws error is searchString is empty', async () => {
     const errorMessage = 'Empty search content'
-    const mockDb = jest.spyOn(PaymentDataModel, 'findAll')
-    mockDb.mockRejectedValue(new Error(errorMessage))
-    await expect(getPaymentData())
+    await expect(getPaymentData({ searchString: '' }))
       .rejects
       .toThrow(errorMessage)
   })
 })
+
+describe('fuzzySearchService tests with sortBy', () => {
+  const searchCriteria = {
+    searchString: 'fa',
+    limit: 5,
+    offset: 0,
+    sortBy: null,
+    filterBy: {
+      schemes: []
+    }
+  }
+
+  const mockDb = jest.spyOn(PaymentDataModel, 'findAll')
+  mockDb.mockResolvedValue(paymentestdata)
+
+  test('GET /paymentdata sorts results by score by default', async () => {
+    const result = await getPaymentData(searchCriteria)
+    expect(result.rows[0].part_postcode).toEqual('PE15')
+  })
+
+  test('GET /paymentdata sorts results by payee name', async () => {
+    const result = await getPaymentData({ ...searchCriteria, sortBy: 'payee_name' })
+    expect(isAscending(result.rows, 'payee_name')).toBe(true)
+  })
+
+  test('GET /paymentdata sorts results by town', async () => {
+    const result = await getPaymentData({ ...searchCriteria, sortBy: 'town' })
+    expect(isAscending(result.rows, 'town')).toBe(true)
+  })
+
+  test('GET /paymentdata sorts results by part_postcode', async () => {
+    const result = await getPaymentData({ ...searchCriteria, sortBy: 'part_postcode' })
+    expect(isAscending(result.rows, 'part_postcode')).toBe(true)
+  })
+
+  test('GET /paymentdata sorts results by county_council', async () => {
+    const result = await getPaymentData({ ...searchCriteria, sortBy: 'county_council' })
+    expect(isAscending(result.rows, 'county_council')).toBe(true)
+  })
+})
+
+describe('fuzzySearchService tests with filterBy', () => {
+  const searchCriteria = {
+    searchString: 'fa',
+    limit: 5,
+    offset: 0,
+    sortBy: null,
+    filterBy: {
+      schemes: []
+    }
+  }
+
+  test('GET /paymentdata returns default results with no schemes provided', async () => {
+    const results = await getPaymentData(searchCriteria)
+    expect(results.count).toBeGreaterThan(0)
+  })
+
+  test('GET /paymentdata filters by single scheme', async () => {
+    const schemes = ['Farming Equipment and Technology Fund']
+    const filteredResult = await getPaymentData({ ...searchCriteria, filterBy: { schemes } })
+
+    filteredResult.rows.forEach(x => {
+      const matchingSet = paymentestdata.find(td => td.payee_name === x.payee_name)
```

Comment:
So that is taking the results returned by getPaymentData endpoint and finding those results in the mockData to check that only results with the matching scheme is returned. You have to do this because the objects in the results will not have a scheme.
However, I didn't expect data with duplicate names but that would be mean that the dataset is invalid? I will double check and amend the data is necessary.

Reply:
duplicate name is possible now right? it could be in the different address. As per current case , we may not have the duplicate name in the same address.

Reply:
Yes, good point. I will compare the post code as well in that case.